exonum/src/messages/protocol.rs

```diff
@@ -503,7 +503,7 @@ macro_rules! impl_protocol {
     };
 }
 
-impl_protocol!{
+impl_protocol! {
```

Comment:
I am not sure that i can make it easier (It's already single-pass, and didn't use external macros, rather than our transactions/encoding_struct).

What kind of layering do you mean? `Services/Responses/Requests` part? I am not sure we can't add new class of messages.

Reply:
I meant explicitly declaring `Protocol` enum without a macro, and using macro only for the repeating code in defining separate message types. These are internal details though, so I'm not particularly persistent about this.

Reply:
It's already defined like you describe. Only methods like deserialise which is based on repetition code defined in macros.

Reply:
> Only methods like deserialise which is based on repetition code defined in macros.

Yeah, I've meant exactly these methods and the declaration of `Protocol` itself, so that you can define the macro like

```rust
impl_protocol! {
    /// Exonum basic node messages.
    Service = 0 {
        /// `RawTransaction` representation.
        RawTransaction = 0,
        /// Handshake to other node.
        Connect = 1,
        /// `Status` information of other node.
        Status = 2,
    },
}
```

Also, I'm having a slight concern that we will need to change the intrinsic of messages if we will implement them in Protobuf.